lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -149,6 +123,125 @@ def remove_schedule(name)
 
       private
 
+      def fallback_setup_schedule(schedule_hash)
+        Resque.logger.warn!("You are running Redis version
+                            #{redis_master_version}. Redis version 2.6 is
+                            required to run Lua scripts to atomically set up
+                            the schedule. resque-scheduler will use the pre 2.6
+                            behaviour which can cause a race condition that
+                            will result in an empty schedule. It is highly
+                            recommended to upgrade Redis for production
+                            workloads.")
```

Comment:
Added tests for logging and fallback method etc
On Apr 14, 2016 3:06 PM, "Simon Eskildsen" <notifications@github.com> wrote:

> In lib/resque/scheduler/scheduling_extensions.rb
> <https://github.com/Shopify/resque-scheduler/pull/7#discussion_r59773533>:
>
> > @@ -149,6 +123,125 @@ def remove_schedule(name)
> >
> >        private
> >
> > +      def fallback_setup_schedule(schedule_hash)
> > +        Resque.logger.warn!("You are running Redis version
> > +                            #{redis_master_version}. Redis version 2.6 is
> > +                            required to run Lua scripts to atomically set up
> > +                            the schedule. resque-scheduler will use the pre 2.6
> > +                            behaviour which can cause a race condition that
> > +                            will result in an empty schedule. It is highly
> > +                            recommended to upgrade Redis for production
> > +                            workloads.")
>
> If I remove this logging, no test fails :upside_down_face:
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Shopify/resque-scheduler/pull/7/files/b3ba8efd878adf9a32fd8a3c2906fef9b7d684dd#r59773533>
>